File: c/alturas.c
#include <stdio.h>

int main() {

    int N, nMenores;
    double soma, media, percentualMenores;

    printf("Quantas pessoas serao digitadas? ");
    scanf("%d", &N);

    char nomes[N][50];
    int idades[N];
    double alturas[N];

    for (int i = 0; i < N; i++) {
        printf("Dados da %da pessoa:\n", i + 1);
        printf("Nome: ");
        fseek(stdin, 0, SEEK_END);
        gets(nomes[i]);
        printf("Idade: ");
        scanf("%d", &idades[i]);
        printf("Altura: ");
        scanf("%lf", &alturas[i]);
    }

    soma = 0;
    for (int i = 0; i < N; i++) {
        soma = soma + alturas[i];
    }
    media = soma / N;
    printf("\nAltura media: %.2llf\n", media);

    nMenores = 0;
    for (int i = 0; i < N; i++) {
        if (idades[i] < 16) {
            nMenores++;
        }
    }
    percentualMenores = nMenores * 100.0 / N;
    printf("Pessoas com menos de 16 anos: %.1lf%%\n", percentualMenores);

    for (int i = 0; i < N; i++) {
        if (idades[i] < 16) {
            printf("%s\n", nomes[i]);
        }
    }

    return 0;
}

File: python/alturas.py
N: int
cont: int
soma: float
media: float
porcentagem: float

N = int(input("Quantas pessoas serão digitadas? "))

nomes: [str] = [0 for x in range(N)]
idades: [int] = [0 for x in range(N)]
alturas: [float] = [0 for x in range(N)]

for i in range(N):
    print(f"Dados da {i + 1}a pessoa:")
    nomes[i] = str(input("Nome: "))
    idades[i] = int(input("Idade: "))
    alturas[i] = float(input("Altura: "))

soma = 0
for altura in alturas:
    soma += altura

media = soma / N
print()
print("Altura media: {:.2f}".format(media))

cont = 0
for i in range(N):
    if idades[i] < 16:
        cont += 1

porcentagem = cont * 100 / N
print("Pessoas com menos de 16 anos: {:.1f}%".format(porcentagem))

print("Nomes das pessoas com menos de 16 anos:")
for i in range(N):
    if idades[i] < 16:
        print(nomes[i])
